src/rgw/rgw_rest_s3.cc

```diff
@@ -527,6 +532,24 @@ int RGWGetObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t bl_ofs,
       decode(rt, iter);
 
       if (rt == rgw::sal::RGWRestoreType::Temporary) {
+        attr_iter = attrs.find(RGW_ATTR_RESTORE_STATUS);
+        if (attr_iter != attrs.end()) {
+          bufferlist bl = attr_iter->second;
+          auto iter = bl.cbegin();
+          decode(restore_status, iter);
```

Comment:
define restore_status and expiration date and other iterators here local to this if loop but not in common structs.